### DISCUSSION TITLE
Custom Image Component

### DISCUSSION THREAD

**Discussion by noor-codes** [keystonejs/keystone#7443] [Questions] [answered]
# Custom Image Component

I want to create an extension of the image component where all files goes into a single table but in the meantime, If it's an image, It should show up as a thumbnail otherwise, it should have a simple link to the data stored in the db.

In the examples repo, we have a custom [json field](https://github.com/keystonejs/keystone/tree/main/examples/json) but the other fields are missing like the images. I found out about this [image](https://github.com/keystonejs/keystone/tree/main/packages/core/src/fields/types/image) component but there are too many internal dependencies, I can't start to customize or copy it to my own project myself.

Does anyone have a custom image component like the json field in the examples repo so I can start customizing it or help me build it myself?

Thank You

**Comment by noor-codes** [comment]
I am also recommended not to move certain files to my own project because they are a part of the system but without them, the custom image component doesn't even run.

![image](https://user-images.githubusercontent.com/28804764/162143447-9d1e9601-601b-432c-b923-cb09cc08c291.png)

**Reply by noor-codes** [reply]
Now, I feel like i'm creating my own version of Keystone which isn't right for sure. I'm doing something the wrong way. 
@mitchellhamilton or @dcousens  Any idea? 🙏

![image](https://user-images.githubusercontent.com/28804764/162145165-3915b689-612e-484f-be04-c5f7d92be876.png)

**Comment by noor-codes** [comment]
In a nutshell, I want to remove this validation to be able to upload any kind of file but render a thumbnail if it's an image otherwise, just a link to the file. 

This way, I can store all things files into a single table, rather than creating different tables for files and images. 

![image](https://user-images.githubusercontent.com/28804764/162169626-f1faf4a2-74ee-4717-8ae5-bfab80bef31f.png)

**Reply by gautamsi** [reply]
you probably need a custom field extend upon the image field. other options
* use `patch-packages` to remove this validation. you probably need more fixes that just these lines
* use a custom view along with file type, you can copy image field view and customize it.

**Reply by noor-codes** [reply]
I want to extend them but the question is: How?

**Reply by noor-codes** [reply]
The `patch-packages` solution seems to be very hacky.

**Reply by gautamsi** [reply]
you can copy the image component `Field` view to start with

**Comment by noor-codes** [comment, marked as answer]
https://github.com/keystonejs/keystone/discussions/7457
